test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
-      const error = new Error('Emission error')
-      transport.on('message', () => {
-        throw error
-      })
+  describe('Event Handling', () => {
+    it('should properly remove event listeners on error', () => {
+      const error = new Error('Test error')
+      transport.emit('error', error)
 
+      // Verify that message and close listeners are removed
       mockEmitter.emit('message', new ArrayBuffer(4))
-      expect(errorListener).toHaveBeenCalledWith(new Error(`Failed to emit message: ${error.message}`))
+      mockEmitter.emit('close')
+
+      expect(errorListener).toHaveBeenCalledWith(error)
+      // No more events should be processed after error
+      expect(mockSocket.send).not.toHaveBeenCalled()
     })
 
-    it('should ignore messages when transport is not active', () => {
+    it('should handle transport cleanup with active timeouts', async () => {
+      const message = new Uint8Array([1])
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
+
+      const sendPromise = transport.sendMessage(message)
+      // Add catch to prevent unhandled rejection
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
+
+      // Process initial attempt
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Close transport while there's a pending timeout
       transport.close()
-      const messageListener = jest.fn()
-      transport.on('message', messageListener)
 
-      mockEmitter.emit('message', new ArrayBuffer(4))
-      expect(messageListener).not.toHaveBeenCalled()
+      // Wait for cleanup to complete
+      await jest.runAllTimersAsync()
+
+      await expect(sendPromise).rejects.toThrow('Connection closed')
+      expect(transport.isConnected).toBe(false)
+
+      // Ensure no more processing occurs
+      await jest.advanceTimersByTimeAsync(5000)
+      expect(mockSocket.send).toHaveBeenCalledTimes(1)
     })
   })
 
-  describe('Connection Management', () => {
-    it('should handle message send when transport is not initialized', async () => {
-      const uninitializedTransport = await createUWebSocketTransport(mockSocket, mockEmitter, mockConfig, mockLogs)
-      // Force transport to be not ready
-      mockSocket.getUserData.mockReturnValue({ isConnected: false })
-      const errorListener = jest.fn()
-      uninitializedTransport.on('error', errorListener)
-
-      const message = new Uint8Array([1])
-      await uninitializedTransport.sendMessage(message)
-      expect(errorListener).toHaveBeenCalledWith(new Error('Transport is not ready or socket is not connected'))
+  describe('Queue Processing Edge Cases', () => {
+    beforeEach(() => {
+      jest.useFakeTimers({ advanceTimers: true })
     })
 
-    it('should handle cleanup during active message processing', async () => {
-      jest.useFakeTimers()
-      mockSocket.send.mockReturnValue(0) // Force send failure
+    afterEach(() => {
+      jest.useRealTimers()
+    })
 
+    // Coverage for lines 200-208
```

Comment:
These comments might get old if we change some lines of code, what do you think about removing it?